### PR TITLE
(PA-3336) add vanagon config for pxp-agent

### PR DESCRIPTION
**Pull request by gimmyxd**

### CODE REVIEW

File: configs/projects/pxp-agent.rb [hunk 1]
```diff
@@ -0,0 +1,68 @@
+project 'pxp-agent' do |proj|
+  platform = proj.get_platform
+  proj.generate_archives true
+  proj.generate_packages false
+
+  # pxp_agent_details = JSON.parse(File.read('configs/components/pxp-agent.json'))
+  # proj.version(pxp_agent_details['ref'])
+  proj.version('test1')
```
Comment (gimmyxd) [inline configs/projects/pxp-agent.rb:8]: TBD: how to handle versions

File: configs/components/cleanup.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    ].each do |component|
+      cleanup_steps << "#{rm} -rf #{settings[:bindir]}/#{component}*"
+    end
+
+    %w[
+      cmake engines pkgconfig ruby libx64-msvcrt yaml-cpp libboost libssl
+      libcrypto libcurl libssl libx64-msvcrt yaml-cpp
+    ].each do |component|
+      cleanup_steps << "#{rm} -rf #{settings[:libdir]}/#{component}*"
+    end
```
Comment (gimmyxd) [inline configs/components/cleanup.rb:37]: Couldn't use `find` successfully on windows.

**Review by ciprianbadescu — APPROVED** [review]

File: configs/components/cleanup.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  cleanup_steps << "#{rm} -rf #{settings[:includedir]}"
+  cleanup_steps << "#{rm} -rf #{settings[:prefix]}/share"
+  cleanup_steps << "#{rm} -rf #{settings[:prefix]}/ssl"
+  cleanup_steps << "#{rm} -rf #{settings[:prefix]}/usr/local"
+  cleanup_steps << "#{rm} -rf #{settings[:prefix]}/CMake"
+
+  if platform.is_windows?
+    %w[
+      erb gem httpclient irb libboost libcrypto libcurl thor yaml-cpp ruby
+      liggcc libconv libssl rake x64-msvcrt-ruby250 rdoc ri rubyw
```
Comment (gimmyxd) [inline configs/components/cleanup.rb:23]: change `x64-msvcrt-ruby250` to `x64-msvcrt-ruby`

File: configs/projects/pxp-agent.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+  proj.license 'See components'
+  proj.vendor 'Puppet, Inc.  <info@puppet.com>'
+  proj.homepage 'https://puppet.com'
+  proj.identifier 'com.puppetlabs'
+
+  # pxp-agent inherits most build settings from puppetlabs/puppet-runtime:
+  # - Modifications to global settings like flags and target directories should be made in puppet-runtime.
+  # - Settings included in this file should apply only to local components in this repository.
+  runtime_details = JSON.parse(File.read('configs/components/puppet-runtime.json'))
+  agent_branch = 'master'
```
Comment (gimmyxd) [inline configs/projects/pxp-agent.rb:20]: change to `main`

File: ext/project_data.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+build_defaults.yaml
```
Comment (GabrielNagy) [inline ext/project_data.yaml:1]: this should be a symlink to ext/build_defaults.yaml but I have no idea who uses it

Comment (gimmyxd) [reply ext/project_data.yaml:1]: it is
i guess this is how git shows it
```
~/Workspace/pxp-agent-vanagon  PA-3336                                                                                           ruby-2.6.3
❯ ls -l ext
total 8
build_defaults.yaml
project_data.yaml -> build_defaults.yaml
```

File: configs/components/puppet-runtime.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  pkg.build_requires 'runtime' if platform.is_cross_compiled_linux? || platform.is_solaris? || platform.is_aix?
+
+  # Even though puppet's ruby comes from puppet-runtime, we still need a ruby
+  # to build with on these platforms:
+  if platform.architecture == 'sparc'
+    pkg.build_requires 'pl-ruby' if platform.os_version == '11'
+  elsif platform.is_cross_compiled_linux?
+    pkg.build_requires 'pl-ruby'
+  end
```
Comment (gimmyxd) [inline configs/components/puppet-runtime.rb:22]: remove this i guess

**Review by GabrielNagy — APPROVED** [review]

**Pull request merged**